### ISSUE TITLE
Support for LCDproc

### ISSUE THREAD

**Issue by dd871** [PeteManchester/MediaPlayer#36]
Hi,

Let me say first that this is a really awesome application! I was searching for weeks for a proper renderer (and surprisingly there are hardly any available at all) and this one ticked all the boxes.

I would really love to have my LCD display what is currently playing through the renderer, unfortunately the plugin does not work, as it is very specific (Raspberry pi with GPIO) and does not match my hardware. I was wondering if it would be possible to support LCDproc instead, which is more of a standard and would enable hundreds of already supported displays to work with MediaPlayer.

The protocol (to implement a LCDproc client) seems to be quite easy from what I understand, it is just text based and connects to the LCD server daemon to issue commands.

More information:
"LCDproc is a piece of open source software that displays real-time system information from your Linux/*BSD box on a LCD. The server supports several serial and USB devices from Matrix Orbital and CrystalFontz as well as some devices connected to the LPT port: HD44780, T6963, SED1520 and SED1330. Various clients are available that display things like CPU load, system load, memory usage, uptime, and a lot more."

http://lcdproc.omnipotent.net/

Hope this could be a possibility.

Best regards,
dd871

**Comment by PeteManchester** [comment]
Hi,

Thanks, LCDproc looks really interesting and a good way to interface with an LCD.
When I get some time I will have a look at building a plugin, it should not be too difficult I already interface with MPD in a similar way.

Thanks again,

Pete.

**Comment by dd871** [comment]
That would be really awesome, thanks for your efforts!

**Comment by PeteManchester** [comment]
Hi,

Just a quick update, I tried installing LCDProc on my rapsi, but could not get it stable with my LCDs, or maybe could not get my LCDs stable with LCDProc.

I had most success installing LCDProc v 0.5.7 as described in this thread

http://forum.kodi.tv/showthread.php?tid=193729

But even though it shows the initial startup message the display soon gets corrupted.

I will try again soon.

Thanks,

Pete.

**Comment by dd871** [comment]
Sounds like you are making good progress though. One thing I missed initially for my display was to change it from "Size=16x2" to "Size=20x2" in the config. Or maybe it is related to the following:

# If your display is slow and cannot keep up with the flow of data from
# LCDd, garbage can appear on the LCDd. Set this delay factor to 2 or 4
# to increase the delays. Default: 1.
#DelayMult=2

Thanks!

**Comment by PeteManchester** [comment]
Hi,

Yes I had tried those options, I have two displays a 20x2 and 20x4 and set the size for each, I also tried setting the delay to 2 and 4 but still got garbage on the display..

Cheers,

Pete.

**Comment by dd871** [comment]
Hi Pete,

its been some time, but I finally had some motivation to have a closer look at how the RPI GPIOs work together with LCDproc. It seems like PI support is standard out of the box (controltype=RPI), so I was wondering is there a particular reason you are going via the pifacecad route? Seems more complicated to me, but maybe you need this to keep it compatible with other things you have running? Anyways, if you should be able to change the controltype to RPI, it will use the hd44780-rpi driver. This driver is setup per default like the following

 * \c rpi connection type of \c hd44780 driver for Hitachi HD44780 based LCD
 * displays connected to the GPIO pins on the Raspberry Pi.
 *
 * The LCD is operated in its 4 bit-mode. R/W (5) on the LCD MUST be hard wired
 * low to prevent 5V logic appearing on the GPIO pins.
 *
 * The default connections are:
 * header(GPIO)   LCD
 * P1-12 (18)     D7 (14)
 * P1-16 (23)     D6 (13)
 * P1-18 (24)     D5 (12)
 * P1-22 (25)     D4 (11)
 * P1-24 (8)      EN (6)
 * GND            R/W (5)
 * P1-26 (7)      RS (4)
 * P1-15 (22)     EN2 (second controller, optional)
 * P1-11 (17)     BL (backlight optiona
 * Mappings can be set in the config file using the key-words:
 * pin_EN, pin_EN2, pin_RS, pin_D7, pin_D6, pin_D5, pin_D4, pin_BL
 * in the [HD44780] section.
 */

Based on your connection diagram your display seems to be already connected in 4-bit mode, so the only thing to change really is to match the GPIO header you connected and add them to the lcdd.conf

If you are still motivated to get your LCD to run with LCDproc (and with that mediaplayer to work with it) then I hope this will help. Let me know if I can assist in any way.

Cheers!

**Comment by PeteManchester** [comment]
Hi,

Sorry I've not had much time to look at this due to work commitments..

When I tested LCDProc I did all of the above steps and it still didn't work, I know the LCD is alright because it work with my own code.

The code itself for the LCDProc should be easy enough t write and a good addition as a plugin, but if it is so difficult to get LCDProc running on the Pi I don't really want to be responsible for supporting people trying to get it working..

Hopefully soon I will get time to test again, if LCDProc is easy to install and works then I can start looking to write the plugin.

Cheers,

Pete.
